File: app/src/main/java/com/zetapush/testsdk/KeyValueCredentialsStorage.java
package com.zetapush.testsdk;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by damien on 20/07/17.
 */

public class KeyValueCredentialsStorage implements StorageCredentialsInterface {

    // Variables
    private final Context context;
    private final String keyLogin;
    private final String keyPassword;

    // Constructor
    public KeyValueCredentialsStorage(Context context, String keyLogin, String keyPassword) {
        this.context = context;
        this.keyLogin = keyLogin;
        this.keyPassword = keyPassword;
    }


    @Override
    public void saveCredentials(String login, String password) {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        editor.putString(keyLogin, login);
        editor.putString(keyPassword, password);
        editor.commit();
    }

    @Override
    public Map<String, String> getCredentials() {
        SharedPreferences sharedPreferences = PreferenceManager.getDefaultSharedPreferences(context);
        String login = sharedPreferences.getString(keyLogin, null);
        String password = sharedPreferences.getString(keyPassword, null);

        Map<String, String> credentials = new HashMap<>();
        credentials.put("login", login);
        credentials.put("password", password);
        return credentials;
    }

    @Override
    public void clearCredentials() {
        this.saveCredentials(null, null);
    }
}

File: android-library/src/main/java/com/zetapush/library/KeyValueCredentialsStorage.kt
package com.zetapush.library

import android.content.Context
import android.content.SharedPreferences
import java.util.*

class KeyValueCredentialsStorage(
    private val context: Context,
    private val businessId: String,
    private val keyLogin: String,
    private val keyPassword: String
) : StorageCredentialsInterface {

    private val sharedPreferencesName: String = "zetapush-shared-pref-$businessId"

    private val sharedPreferences: SharedPreferences
        get() = context.getSharedPreferences(sharedPreferencesName, Context.MODE_PRIVATE)

    override fun saveCredentials(login: String?, password: String?) {
        val editor = sharedPreferences.edit()
        login?.let { login ->
            editor.putString(keyLogin, login)
        }
        password?.let { password ->
            editor.putString(keyPassword, password)
        }
        editor.apply()
    }

    override val credentials: Credentials?
        get() {
            val login = sharedPreferences.getString(keyLogin, null)
            val password = sharedPreferences.getString(keyPassword, null)

            return Credentials(login, password)
        }

    override fun clearCredentials() {
        val edit = sharedPreferences.edit()
        sharedPreferences.all.forEach { edit.remove(it.key) }
        edit.apply()
    }
}
